include/swift/AST/TypeCheckRequests.h

```diff
@@ -91,10 +91,11 @@ class InheritedTypeRequest :
 };
 
 /// Request the superclass type for the given class.
-class SuperclassTypeRequest :
-    public SimpleRequest<SuperclassTypeRequest,
-                         Type(NominalTypeDecl *, TypeResolutionStage),
-                         CacheKind::SeparatelyCached> {
+class SuperclassTypeRequest
+    : public SimpleRequest<
+          SuperclassTypeRequest, Type(NominalTypeDecl *, TypeResolutionStage),
+          CacheKind::SeparatelyCached | CacheKind::DependencySink |
+              CacheKind::DependencySource> {
```

Comment:
Why not, eg, SuperclassDeclRequest?

Reply:
Either one works. I chose to instrument the one that we execute while type checking the primary.